### PR TITLE
FIX: weightnorm variables

### LINKED ISSUE #203: WeightNormalization test are failing without symbolic tensors in build
Since https://github.com/tensorflow/tensorflow/commit/9df99aafc3480638cd8d2bf6a0caeab111b68302 our keras_layer test is failing on weightnorm. 

This is the block that fails from the [test uti](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/testing_utils.py#L207)l:
```
 # test serialization, weight setting at model level
  model_config = model.get_config()
  recovered_model = keras.models.Sequential.from_config(model_config)
  if model.weights:
    weights = model.get_weights()
    recovered_model.set_weights(weights)
    output = recovered_model.predict(input_data)
    np.testing.assert_allclose(output, actual_output, rtol=2e-3
```
So our model config isn't properly serializing the data. 

@omalleyt12 could you take a look and provide some insight. I'm not overly familiar with `ops.init_scope` and the check for `_keras_style`

Here is our weight norm layer:
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/wrappers.py#L24

### LINKED ISSUE #216: WeightNormalization fails after update
Yesterday I cut a 0.3 release on pypi test, but as a sanity check on the weightnorm changes #208 I ran a quick regression check:

https://colab.research.google.com/drive/1RQlVR9X7vj8q3W3sJC6YA3Cq45XTXrau

The new layer is failing with loss of nan. Looking at the test coverage for this layer I'm struggling how to best catch this without running a full model. The losses for our toy test cases do not return nan so perhaps we need a true regression test like mnist?

EDIT -- Fix colab link

### PR DESCRIPTION
**Pull request by seanpmorgan**
Closes #216

Looks like assigning the variables using operations during init created an issue in the gradients. Setting their values inside of tf.function as done before appears to fix it. I'm not really thrilled about the lack of test coverage, but also not excited to build a regression test into our unit tests. 

https://colab.research.google.com/drive/1Md7SnyEC5bUfkME1Akth4KKM24ac26NK

After this bugfix is in I'll work on publishing a 0.3 release

### CODE REVIEW

**Comment by facaiy** [comment]
Thanks for ping me, Sean, I'll look into it tomorrow. By the way, is it related with https://github.com/tensorflow/addons/pull/220 ?

**Comment by WindQAQ** [comment]
> Thanks for ping me, Sean, I'll look into it tomorrow. By the way, is it related with #220 ?

No, it's just the typo in example :-)

**Comment by seanpmorgan** [comment]
> Thanks for ping me, Sean, I'll look into it tomorrow. By the way, is it related with #220 ?

No problem that was auto from CODEOWNERS. @qlzh727 might want to review as this was just worked on.

**Changes:** 
1. `data_init` also calls `init_norm` because it was previously called in `build`
2. `init_norm` now calls `_compute_weights` as it ran afterward in `build` before
2. The variable assigns were causing broken gradients.

**Comment by qlzh727** [comment]
I am ok with point 3 which changes the .assign() to =, but I am still bit confused about 1 and 2. 

a. Both data init and init_norm are trying to init the value of g, and I don't see the reason to make init_norm to call _compute_weights(), which will update the self.layer.kernel. 
b. For my understanding, init_norm and data_init are trying to do the same thing, which is initialize the g value. In the data init's case, does it also require init_norm()?

**Comment by seanpmorgan** [comment]
> I am ok with point 3 which changes the .assign() to =, but I am still bit confused about 1 and 2.
> 
> a. Both data init and init_norm are trying to init the value of g, and I don't see the reason to make init_norm to call _compute_weights(), which will update the self.layer.kernel.
> b. For my understanding, init_norm and data_init are trying to do the same thing, which is initialize the g value. In the data init's case, does it also require init_norm()?

Ah sry thats a bit of laziness on my part. During troubleshooting I just made everything match the same steps as it was before to try to debug. I've removed 1+2 and confirmed the expected results:
https://colab.research.google.com/drive/1Md7SnyEC5bUfkME1Akth4KKM24ac26NK

**Comment by facaiy** [comment]
Is it a bug? I think we are encouraged to use `assign` than `=`, if I'm not wrong. cc @alextp

**Comment by alextp** [comment]
using `assign` sets the value of a tf.Variable, using `=` will set the value of a python variable to be that tensor. Assigning to tf.Variables is not differentiable but making a tensor based on some operations is.

Doing `self.bias = ...` inside a tf.function seems very dangerous because you've now kept a reference to a tensor defined inside a function which you're not allowed to use outside the function.

I recommend this code be changed to just say `bias = ` instead of `self.bias = `. Then I believe it is correct.
